### PR TITLE
Update workflows.adoc

### PR DESCRIPTION
**Pull request by nathanfishcircleci**
# Description
Added terminal state concepts and note about non-terminal states getting cancelled after 90 days. 

NOTE: NEEDS Review from Stig! 

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/workflows.adoc
Comment (stig) [inline jekyll/_cci2/workflows.adoc:47]: NOT RUN is a terminal state; there is no other state it can transition to from there.

```suggestion
| Yes
```

File: jekyll/_cci2/workflows.adoc
Comment (stig) [inline jekyll/_cci2/workflows.adoc:65]: ~ just a note that in the backend and APIs this state is `ON_HOLD`, but the UI started renaming it as of 2 months ago. Should we include both statuses? Or a separate column for the UI status?

```suggestion
| NEEDS APPROVAL (UI) / ON HOLD
```

File: jekyll/_cci2/workflows.adoc
Comment (stig) [inline jekyll/_cci2/workflows.adoc:72]: `NEEDS SETUP` is not actually a workflow state we support. From the description I think what is meant here is `ERROR`, which is a terminal status. I've searched the UI to verify, and it does not look like this is a case of the UI renaming the `ERROR` state to `NEEDS SETUP` (like it does for `ON HOLD` -> `NEEDS APPROVAL`)

```suggestion
| ERROR
| A workflow stanza is not included or is incorrect in the xref:configuration-reference#[`.circleci/config.yml` file] for this project
| Yes

|
| UNAUTHORIZED
| One or more of the jobs terminated with a `unauthorized` job status, because the triggerer does not have access to a required restricted context.
| Yes
|===
```

File: jekyll/_cci2/workflows.adoc
Comment (stig) [inline jekyll/_cci2/workflows.adoc:74]:
```suggestion
NOTE: After 90 days non-terminal workflows will be cancelled by the CircleCI Automation actor.
```

File: jekyll/_cci2/workflows.adoc
Comment (stig) [inline jekyll/_cci2/workflows.adoc:54]: Maybe clarify a bit more the difference between this and `failed`

```suggestion
| A job in the workflow has failed, but others are still running or yet to be approved
```

**Review by stig — CHANGES_REQUESTED** [review]

File: jekyll/_cci2/workflows.adoc
Comment (rosieyohannan) [inline jekyll/_cci2/workflows.adoc:39]:
```suggestion
| State | Description | Terminal state
```

File: jekyll/_cci2/workflows.adoc
Comment (stig) [inline jekyll/_cci2/workflows.adoc:70]: It occurred to me that config issues are now caught higher up the stack. We'll get an error at the pipeline level if there are issues with the validity of the config. I had another look at this and the only time we set an `ERROR` status for workflows is when give up processing an `unblock-jobs` message after 10 minutes of retries, which in the majority cases is due to a failure building the permissions profile. It is quite rare.

```suggestion
| We experienced an internal error starting a job in the workflow
```

**Review by stig — APPROVED** [review]

**Comment by stig** [comment]
My concerns have been addressed. The issues the linter is bleating about seems to be unrelated to this PR.

**Pull request merged**
